### PR TITLE
Extend reservation endpoint

### PR DESCRIPTION
**Pull request by igordavydsson**
- Add include parameter to GET /reservation/ endpoint

Refs VAR-140

### CODE REVIEW

File: resources/api/reservation.py [hunk 1]
```diff
@@ -502,9 +512,12 @@ class ReservationViewSet(munigeo_api.GeoModelAPIView, viewsets.ModelViewSet, Res
         .prefetch_related('catering_orders').prefetch_related('resource__groups').order_by('begin', 'resource__unit__name', 'resource__name')
 
     serializer_class = ReservationSerializer
-    filter_backends = (DjangoFilterBackend, filters.OrderingFilter, UserFilterBackend, ReservationFilterBackend,
-                       NeedManualConfirmationFilterBackend, StateFilterBackend, CanApproveFilterBackend)
+    filter_backends = (DjangoFilterBackend, filters.OrderingFilter, filters.SearchFilter, UserFilterBackend,
+                       ReservationFilterBackend, NeedManualConfirmationFilterBackend, StateFilterBackend,
+                       CanApproveFilterBackend)
     filterset_class = ReservationFilterSet
+    search_fields = ('user__first_name', 'user__last_name', 'user__email',
```
Comment (jussih) [inline resources/api/reservation.py]: What's the use case for the generic search filter? Currently the search filters for Reservation are in ReservationFilterSet (https://github.com/City-of-Helsinki/respa/blob/master/resources/api/reservation.py#L400) and partially overlapping what this new `search` filter is doing.

Can the use case be satisfied through ReservationFilterSet?

Comment (igordavydsson) [reply resources/api/reservation.py]: ah, okay, I think I can extend `ReservationFilterSet` then, thanks! 🙂

File: resources/api/reservation.py [hunk 2]
```diff
@@ -115,6 +116,15 @@ def __init__(self, *args, **kwargs):
             for field_name in required:
                 self.fields[field_name].required = True
 
+    def get_extra_fields(self, includes, context):
+        from .resource import ResourceSerializer
+
+        """ Define extra fields that can be included via query parameters. Method from ExtraDataMixin."""
+        extra_fields = {}
+        if 'resource' in includes:
```
Comment (jussih) [inline resources/api/reservation.py]: `ResourceSerializer` has used `unit_detail` as the key to get the full unit data included, use the same notation here.

File: resources/api/reservation.py (hunk 1 above)
Comment (igordavydsson) [reply resources/api/reservation.py]: This will be introduced in a separate PR.

File: resources/api/reservation.py (hunk 2 above)
Comment (igordavydsson) [reply resources/api/reservation.py]: changed the keyword

**Comment by igordavydsson** [comment]
Okay, thanks @tuomas777 and @jussih ! 🙌 I took a different approach to the problem now. `ReservationViewSet` has [`.select_related('resource', 'resource__unit')`](https://github.com/City-of-Helsinki/respa/blob/master/resources/api/reservation.py#L501), so performance should not be an issue for the specified fields.

**Pull request merged**